src/Voron/Slice.Extensions.cs

```diff
@@ -18,11 +18,7 @@ public static bool StartWith(this Slice s1, ReadOnlySpan<byte> s2)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]        
         public static bool EndsWith(this Slice s1, ReadOnlySpan<byte> s2)
         {
-            var term = s1.AsReadOnlySpan();
-            
-            return term.Length > 1 
-                ? s1.AsReadOnlySpan()[..(term.Length - 1)].EndsWith(s2)
-                : s1.AsReadOnlySpan().EndsWith(s2);
+            return s1.AsReadOnlySpan().EndsWith(s2);
```

Comment:
Like above. Did you handle that case?

Reply:
For example when you have indexed `term`.
So, s1 is written as `term\0` but s2 is `term`.

Reply:
There is no reason why you should be doing that here... I dont know what I was thinking when I wrote this code. Compact Tree development era.